src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -143,54 +143,82 @@ def report_json(
     get_logger().info(f"reporting the messages: {msgs[:10]}")
     try:
         prune_trace: bool = not os.environ.get("LUMIGO_PRUNE_TRACE_OFF", "").lower() == "true"
-        to_send = _create_request_body(msgs, prune_trace).encode()
+        should_try_zip: bool = _should_try_zip()
+        to_send: Union[str, List[str]] = _create_request_body(
+            region, msgs, prune_trace, should_try_zip
+        )
     except Exception as e:
         get_logger().exception("Failed to create request: A span was lost.", exc_info=e)
         return 0
     if should_use_tracer_extension():
         with lumigo_safe_execute("report json file: writing spans to file"):
             write_spans_to_files(spans=msgs, is_start_span=is_start_span)
         return 0
-    if region == CHINA_REGION:
-        return _publish_spans_to_kinesis(to_send, CHINA_REGION)
-    host = None
+    if region == CHINA_REGION and isinstance(to_send, str):
+        return _publish_spans_to_kinesis(to_send.encode(), CHINA_REGION)
+
     global edge_connection
     with lumigo_safe_execute("report json: establish connection"):
         host = get_edge_host(region)
         duration = 0
         if not edge_connection or edge_connection.host != host:
             edge_connection = establish_connection(host)
             if not edge_connection:
-                get_logger().warning("Can not establish connection. Skip sending span.")
+                get_logger().warning("Cannot establish connection. Skip sending span.")
                 return duration
+
     try:
         start_time = time.time()
+
+        to_send = to_send if isinstance(to_send, list) else [to_send]
+        get_logger().debug(f"Going to send a list of {len(to_send)} spans...")
+        # Process each span one by one
+        for span_data in to_send:
+            send_single_span(host, span_data)
```

Comment:
note, the `to_send` var is a list of http request bodies to send... not list of spans.

When not zipping the `to_send` contains one request with all the spans to send, and when zipping it can be a list of requests to send.

Please fix the name, comment and log msg

Reply:
Done